### PR TITLE
Updated the shopping example

### PR DESCRIPTION
**Pull request by HansMuller**

### CODE REVIEW

**Comment by qchong** [comment]
Lgtm

File: widgets-intro.md [hunk 1]
```diff
@@ -469,17 +469,19 @@ class ShoppingListItem extends StatelessWidget {
...
   TextStyle _getTextStyle(BuildContext context) {
-    if (inCart) {
-      return DefaultTextStyle.of(context).copyWith(
-          color: Colors.black54, decoration: TextDecoration.lineThrough);
-    }
-    return null;
+    if (!inCart)
+      return null;
+
+    return new TextStyle(
```
Comment (Hixie) [inline widgets-intro.md]: this is a semantic change, is it intentional?

Comment (Hixie) [inline widgets-intro.md]: oh i see, the Text merges this style in with the inherited style anyway. ok.

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
You may have to fix the web site first. Looks like something broke.

File: widgets-intro.md (hunk 1 above)
Comment (HansMuller) [inline widgets-intro.md]: Right. I was pleased to be able to remove the DefaultTextStyle copyWith() code. Not something you want to see in an intro :-).

**Comment by HansMuller** [comment]
Any idea how to run the "Extract Dart snippets from the markdown documentation" step? I can't figure out what our lint pass is complaining about:

```
17 code snippets extracted.
Analyzing the extracted Dart libraries.
Analyzing 35 files...
[lint] Specify type annotations. (tool/extract.dart, line 20, col 15)
[lint] Specify type annotations. (tool/extract.dart, line 21, col 18)
[lint] Specify type annotations. (tool/extract.dart, line 98, col 15)
[lint] Specify type annotations. (tool/extract.dart, line 99, col 18)
```

**Comment by devoncarew** [comment]
It looks like it's complaining about the extraction tool itself.

**Pull request merged**